### DISCUSSION TITLE
Why can't `matchUpdateTypes` & `separateMultipleMinor` be used in the same package rule?

### DISCUSSION THREAD

**Discussion by SchroederSteffen** [renovatebot/renovate#29064] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub Enterprise

### Please tell us more about your question or problem

Please see the reproduction repo: <https://github.com/SchroederSteffen/renovate-matchUpdateTypes-separateMultipleMinor?#readme>.

### Problem

In our org, we group all minor & patch updates (preset `group:allNonMajor`). However, we want to separate any minor TypeScript updates, because they may contain breaking changes and many tools restrict the supported minor versions.

Additionally, we want to use the option `separateMultipleMinor` to be able to perform version jumps one-by-one.

Unfortunately, Renovate doesn't allow using `matchUpdateTypes` & `separateMultipleMinor` in the same package rule because of: <https://github.com/renovatebot/renovate/blob/8d78ca2ec8413739ac5c9247393ef8a147bfbd80/lib/config/validation.ts#L490>

### Workaround

While it's not possible to use `matchUpdateTypes` & `separateMultipleMinor` in the same package rule, you can write two package rules as a workaround.

```json
{
  "packageRules": [
    {
      "groupName": "typescript",
      "matchPackageNames": ["typescript"],
      "matchUpdateTypes": ["major", "minor"]
    },
    {
      "matchPackageNames": ["typescript"],
      "separateMultipleMinor": true
    }
  ]
}
```

## Question

Could you please confirm that this workaround is necessary?
Assuming this kind of a restriction based on the current implementation, Is there some documentation explaining this?

### Thanks

In any case, many thanks for providing Renovate and all the support around it! ❤️

**Comment by rarkins** [comment, marked as answer]
It's not a workaround - it's the solution.

separateMultipleMinor is something which needs to be set before Renovate decides if an update is minor or patch. Then matchUpdateTypes is something which happens after. If you match on update types then it's too late

**Reply by SchroederSteffen** [reply]
Thanks for the response.
It's just that as a user that doesn't know the internals, I kind of expected that a single rule should be enough.
I'm okay with this solution, though.
